helper-cli/src/main/kotlin/commands/GeneratePackageConfigurationsCommand.kt

```diff
@@ -77,11 +85,28 @@ internal class GeneratePackageConfigurationsCommand : CliktCommand(
 
     private fun Provenance.writePackageConfigurationFile(filename: String) {
         val packageConfiguration = createPackageConfiguration(packageId, this)
-        val outputFile = outputDir.resolve(filename)
+        val outputFile = getOutputFile(filename)
 
         packageConfiguration.writeAsYaml(outputFile)
         println("Wrote a package configuration to '${outputFile.absolutePath}'.")
     }
+
+    private fun getOutputFile(filename: String): File {
+        val relativeOutputFilePath = buildString {
+            fun appendPathElement(pathElement: String) = append(pathElement.ifBlank { "_" }).append("/")
+
+            if (createHierarchicalDirs) {
+                appendPathElement(packageId.type)
```

Comment:
Shouldn't we try to make reuse of `Identifier.toPath()` here?

Reply:
Good point. I've generalized it first to make use of it.